yt/frontends/owls/tests/test_outputs.py

```diff
@@ -54,3 +55,23 @@ def test_snapshot_033():
 @requires_file(os33)
 def test_OWLSDataset():
     assert isinstance(data_dir_load(os33), OWLSDataset)
+    
+    
+@requires_ds(os33)
+def test_OWLS_particlefilter(os33):
+    ds = data_dir_load(os33)
+    ad = ds.all_data()
+    def cold_gas(pfilter, data):
+        temperature = data[pfilter.filtered_type, "Temperature"]
+        filter = (temperature.in_units('K') <= 1e5)
+        return filter
+    add_particle_filter("gas_cold", function=cold_gas, filtered_type='PartType0', requires=["Temperature"])
+    ds.add_particle_filter('gas_cold')
+
+    mask = (ad['PartType0','Temperature'] <= 1e5)
+    try:
```

Comment:
I would replace this whole try/except block with the following line:

```
assert ad['PartType0','Temperature'][mask].shape == ad['gas_cold','Temperature'].shape
```

There's no need for fancy error handling in a test, if the test breaks we'd like to know where it breaks :)